src/components/verification-grid/verification-grid.ts

```diff
@@ -944,20 +930,21 @@ export class VerificationGridComponent extends AbstractComponent(LitElement) {
       throw new Error("No paginator found.");
     }
 
-    this.subjectHistory.push(...this.currentPage);
-    this.currentPage = await this.paginationFetcher.getItems(count);
-    this.verificationBuffer = this.currentPage;
-    await this.renderVirtualPage(this.currentPage);
+    const nextBuffer = await this.paginationFetcher.getItems(count);
```

Comment:
hmm one of the points of this change is that the fetching of pages of subjects was meant to be decoupled from paging of the grid.

So, in the background we can fetch thousands of subjects asynchronously.

The subject pager also shouldn't even accept a count of items to fetch. The ecosounds APi for example ideally fetches 25 items at a time. That has little or nothing to do with the grid size

Reply:
Discussed offline

getPageFetcher currently has an internal array of subjects

Since we now have a combined subjects array, we should store pre-fetched subjects in this array as well

---

Changes

- getPageFetcher will still exist, but it will be a class with async methods that works over the verification grid's subject buffer
    - `populateSubjects` will populate the verification grids subjects array until we reach the pre-fetch limit
    - `cacheAudio` will async iterate over the the verification grids subjects array checking if the audio has been cached. If not, it will do the appropriate GET or HEAD request for the audio
        - cacheAudio will be automatically run if populateSubjects results in more items being appended to the subjects buffer